### PR TITLE
Some bugfixes to prepare for release

### LINKED ISSUE #244: Ladder teleport icon still visible in old mine @JucanAndreiDaniel

### PR DESCRIPTION
**Pull request by JucanAndreiDaniel**
Fixes #244 #245 #246 
Also includes
1. fix for weird raycast when teleporting (colliding with music zone or teleport trigger)
2. fix error when testing with sounds deactivated

# Testing
* [x] Tested with PCVR
* [x] Tested with Pico / Quest

### CODE REVIEW

File: Assets/GothicVR/Resources/Prefabs/UI Element Prefabs/MainMenu.prefab [hunk 1]
```diff
@@ -3636,7 +3636,7 @@ MonoBehaviour:
   m_EditorClassIdentifier: 
   m_Material: {fileID: 0}
   m_Color: {r: 1, g: 1, b: 1, a: 1}
-  m_RaycastTarget: 0
+  m_RaycastTarget: 1
```
Comment (JaXt0r) [inline Assets/GothicVR/Resources/Prefabs/UI Element Prefabs/MainMenu.prefab:3639]: Just to double check: Intended?

File: Assets/GothicVR/Resources/Prefabs/UI Element Prefabs/MainMenu.prefab [hunk 2]
```diff
@@ -8768,7 +8768,7 @@ MonoBehaviour:
   m_PrefabInstance: {fileID: 0}
   m_PrefabAsset: {fileID: 0}
   m_GameObject: {fileID: 9044881996792586950}
-  m_Enabled: 0
+  m_Enabled: 1
```
Comment (JaXt0r) [inline Assets/GothicVR/Resources/Prefabs/UI Element Prefabs/MainMenu.prefab:8771]: Just to double check: Intended?

File: Assets/GothicVR/Scripts/Creator/VobCreator.cs [hunk 3]
```diff
@@ -480,7 +483,9 @@ private static GameObject CreateZoneMusic(PxVobZoneMusicData vob)
             var go = PrefabCache.TryGetObject(PrefabCache.PrefabType.VobMusic);
             go.SetParent(parentGosNonTeleport[vob.type], true, true);
             go.name = vob.vobName;
-            
+
+            go.layer = ConstantsManager.IgnoreRaycastLayer;
```
Comment (JaXt0r) [inline Assets/GothicVR/Scripts/Creator/VobCreator.cs:487]: Do we really need a separate Layer for this? Afaik Layers have other side effects like colliders won't work etc.

Comment (JaXt0r) [reply Assets/GothicVR/Scripts/Creator/VobCreator.cs:487]: This one might not need a collider, sure. But the question is more like: Do you know the drawbacks and is it really needed?

File: Assets/GothicVR/Scripts/Manager/GvrSceneManager.cs [hunk 4]
```diff
@@ -166,14 +166,12 @@ private async Task ShowLoadingScene(string worldName = null)
             await Task.Delay(ensureLoadingBarDelayMilliseconds);
         }
 
-        private void SetLoadingTextureForWorld(string worldName)
+        private void SetLoadingTextureForWorld(string worldName, bool newGame = false)
```
Comment (JaXt0r) [inline Assets/GothicVR/Scripts/Manager/GvrSceneManager.cs:169]: You name newGame as parameter, but don't handle it inside this function. Are you missing something?

File: Assets/GothicVR/Scripts/Player/Climb/RayTeleportClimb.cs [hunk 5]
```diff
@@ -150,6 +150,7 @@ private void CreateReticle()
             reticleImage.type = Image.Type.Filled;
             reticleImage.fillMethod = Image.FillMethod.Vertical;
             Destroy(reticle.GetComponent<Collider>());
+            reticle.SetActive(false);
```
Comment (JaXt0r) [inline Assets/GothicVR/Scripts/Player/Climb/RayTeleportClimb.cs:153]: I know it's not the topic for your branch, but can we please rename this one to something meaningful? ;-)

**Review by JaXt0r — CHANGES_REQUESTED** [review]
A few questions only.

File: Assets/GothicVR/Resources/Prefabs/UI Element Prefabs/MainMenu.prefab (hunk 1 above)
Comment (JucanAndreiDaniel) [reply Assets/GothicVR/Resources/Prefabs/UI Element Prefabs/MainMenu.prefab:3639]: yes

File: Assets/GothicVR/Resources/Prefabs/UI Element Prefabs/MainMenu.prefab (hunk 2 above)
Comment (JucanAndreiDaniel) [reply Assets/GothicVR/Resources/Prefabs/UI Element Prefabs/MainMenu.prefab:8771]: yes

File: Assets/GothicVR/Scripts/Manager/GvrSceneManager.cs (hunk 4 above)
Comment (JucanAndreiDaniel) [reply Assets/GothicVR/Scripts/Manager/GvrSceneManager.cs:169]: did you mean use it? if so check line 174

File: Assets/GothicVR/Scripts/Creator/VobCreator.cs (hunk 3 above)
Comment (JucanAndreiDaniel) [reply Assets/GothicVR/Scripts/Creator/VobCreator.cs:487]: this is the collider for music, we need to know when to change the track
it is intended to set the layer to this as we might get some weird interactions between the teleport raycast and collider

File: Assets/GothicVR/Scripts/Manager/GvrSceneManager.cs (hunk 4 above)
Comment (JaXt0r) [reply Assets/GothicVR/Scripts/Manager/GvrSceneManager.cs:169]: Too much beer. Sorry. Solved. 🤣

File: Assets/GothicVR/Scripts/Creator/VobCreator.cs (hunk 3 above)
Comment (JaXt0r) [reply Assets/GothicVR/Scripts/Creator/VobCreator.cs:487]: Ok. Agreed. Can we at least add a comment to the ConstantsManager where we use this Layer, please. I have a feeling in 6 months we forget why we have an IgnoreRaycastLayer. Thanks.

**Review by JaXt0r — APPROVED** [review]
Nice! Code looks fine.
Let's wait for Romeyyy's functional testings.

**Pull request merged**